relayer/relayer/src/contracts/azero.rs

```diff
@@ -125,44 +120,12 @@ impl MostInstance {
             bytes32_to_str(&dest_receiver_address),
             request_nonce.to_string(),
         ];
-        let call_data = self.transcoder.encode("receive_request", args)?;
-
-        let dry_run_args = ContractCallArgs {
-            origin: signed_connection.account_id().clone(),
-            dest: self.address.clone(),
-            value: 0,
-            gas_limit: Some(gas_limit.clone()),
-            storage_deposit_limit: None,
-            input_data: call_data.clone(),
-        };
-
-        // Dry run to detect potential errors
-        let dry_run_res = match signed_connection.call_and_get(dry_run_args).await?.result {
-            Ok(res) => res,
-            Err(why) => {
-                error!("Dry run failed: {:?}", why);
-                return Err(AzeroContractError::DispatchError(format!("{:?}", why)));
-            }
-        };
-        if dry_run_res.did_revert() {
-            let decoded_value = self
-                .transcoder
-                .decode_return("receive_request", &mut dry_run_res.data.as_ref());
+        let params = ExecCallParams::new().gas_limit(gas_limit);
 
-            error!("Dry run reverted: {:?}", decoded_value);
-
-            return Err(AzeroContractError::DryRunReverted(decoded_value));
-        }
-
-        let call_result = signed_connection
-            .call(
-                self.address.clone(),
-                0,
-                gas_limit,
-                None,
-                call_data,
-                TxStatus::Finalized,
-            )
```

Comment:
I have quite a strong opinion for migrating to subxt eventually, so I can leave explicit dry-running and `call` instead of the higher level `exec` from aleph-client, to facilitate this in the future. Wdyt?